internal/pkg/warewulfd/parser.go

```diff
@@ -44,8 +47,17 @@ func parseReq(req *http.Request) (parserInfo, error) {
 		ret.efifile = path_parts[2]
 	}
 	ret.hwaddr = hwaddr
+    ipaddrtemp = ""
+    ipaddr = strings.Split(req.RemoteAddr, ":")
+    // IPv6 addresses hextets are split on ":"
+    if len(ipaddr) > 2 {
+       for i := 0; i < len(ipaddr)-1; i++ {
+            ipaddrtemp += ipaddr[i] + ":"
+       }
+       ret.ipaddr6 = strings.TrimSuffix(ipaddrtemp, ":")
+    }
```

Comment:
I think this would be simpler with something like

```
parts := string.Split(req.RemoteAddr, ":")
if len(parts) > 2 {
    ret.ipaddr6 = strings.Join(parts[:len(parts)-1], ":")
}
```